freqtrade/plugins/pairlist/VolumePairList.py

```diff
@@ -103,6 +141,53 @@ def filter_pairlist(self, pairlist: List[str], tickers: Dict) -> List[str]:
         # Use the incoming pairlist.
         filtered_tickers = [v for k, v in tickers.items() if k in pairlist]
 
+        # get lookback period in ms, for exchange ohlcv fetch
+        if self._use_range:
+            since_ms = int(arrow.utcnow()
+                           .floor('minute')
+                           .shift(minutes=-(self._lookback_period * self._tf_in_min)
+                                  - self._tf_in_min)
+                           .int_timestamp) * 1000
+
+            to_ms = int(arrow.utcnow()
+                        .floor('minute')
+                        .shift(minutes=-self._tf_in_min)
+                        .int_timestamp) * 1000
+
+            # todo: utc date output for starting date
+            self.log_once(f"Using volume range of {self._lookback_period} candles, timeframe: "
+                          f"{self._lookback_timeframe}, starting from {format_ms_time(since_ms)} "
+                          f"till {format_ms_time(to_ms)}", logger.info)
+            needed_pairs = [
+                (p, self._lookback_timeframe) for p in
+                [
+                    s['symbol'] for s in filtered_tickers
+                ] if p not in self._pair_cache
+            ]
+
+            # Get all candles
+            candles = {}
+            if needed_pairs:
+                candles = self._exchange.refresh_latest_ohlcv(
+                    needed_pairs, since_ms=since_ms, cache=False
+                )
+            for i, p in enumerate(filtered_tickers):
+                pair_candles = candles[
+                    (p['symbol'], self._lookback_timeframe)
+                ] if (p['symbol'], self._lookback_timeframe) in candles else None
+                # in case of candle data calculate typical price and quoteVolume for candle
+                if pair_candles is not None and not pair_candles.empty:
+                    pair_candles['typical_price'] = (pair_candles['high'] + pair_candles['low']
+                                                     + pair_candles['close']) / 3
+                    pair_candles['quoteVolume'] = (
+                        pair_candles['volume'] * pair_candles['typical_price']
+                    )
+
+                    # replace quoteVolume with range quoteVolume sum calculated above
+                    filtered_tickers[i]['quoteVolume'] = pair_candles['quoteVolume'].sum()
```

Comment:
Wouldn't you want to use something like `pair_candles['quoteVolume'].rolling(self._lookback_period.sum()` - and use the last value of that result?

while `since_ms` is used to specify since when we'd like to have history - i'm not certain it'll work reliably for all exchanges (depends a bit on the exchange api functioning).
if you provide a since_ms very close to the current date (closer than `now - minimum candles`  - this might result in using 500 candles even though you only specified 20 ...

Reply:
fair point! didn't have in mind that a very tight delta between since_ms and now can result in fetching the maximum candle amount.

Reply:
doesn't even have to e very tight - some exchanges provide you a minimum of 100 candles, even if you only wanted 20 ...

Reply:
yeah, building a total sum with a varying amount of candles depending on how the exchange behaves was stupid anyway, so this is definitely a much more safer approach 👌